(GH-2078) Add Bolt guide pages (#2088)
puppetlabs/bolt: guides/project.txt

TOPIC
    project

DESCRIPTION
    A Bolt project is a directory that serves as the launching point for Bolt
    and allows you to create a shareable orchestration application. Projects
    typically include a project configuration file, an inventory file, and any
    content you use in your project workflow, such as tasks and plans.

    When you run Bolt, it runs in the context of a project. If the directory you
    run Bolt from is not a project, then Bolt will attempt to find a project by
    traversing the parent directories. If Bolt is unable to find a project, it
    will run from the default project, located at '~/.puppetlabs/bolt'.

>     A directory is only considered a Bolt project when it has a project
>     configuration file named 'bolt-project.yaml'. Project data and content,
>     including inventory files, are not loaded by Bolt unless they are part of
>     a project.

Reviewer: ```suggestion
    A directory is only considered a Bolt project when it has a project
    configuration file named 'bolt-project.yaml'. Bolt doesn't load project data 
    and content, including inventory files, unless the data and content are 
    part of a project.
```
